crates/core/tedge_write/src/api.rs

```diff
@@ -57,7 +59,11 @@ impl<'a> CopyOptions<'a> {
     }
 
     fn command(&self) -> anyhow::Result<Command> {
-        let mut command = self.sudo.command(crate::TEDGE_WRITE_PATH);
+        // if tedge-write is in PATH of tedge process, use it, if not, defer PATH lookup to sudo
+        let tedge_write_binary =
+            which::which_global(TEDGE_WRITE_BINARY).unwrap_or(TEDGE_WRITE_BINARY.into());
+
+        let mut command = self.sudo.command(tedge_write_binary);
```

Comment:
Oops, I have not been fast enough to answer @albinsuresh question before this PR being merged by @Bravo555 !

> I'm just wondering if running all sudo commands issued by `tedge-agent` with the `--preserve-env` option would be a better fix, as that'll make sure that not just `tedge-write` but any other process spawned by `tedge-agent`, like the software management plugins as well, would benefit from the same fix. I'm only concerned if it's some sort of "bad security practice" invoking `sudo` preserving the calling process's env.

One simple way,  to have all the components benefiting from the same fix, is to move the call to `which::which_global()`  *inside* the method `sudo.command()`.